analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -62,20 +66,27 @@ import org.ossreviewtoolkit.utils.await
 import org.ossreviewtoolkit.utils.collectMessagesAsString
 import org.ossreviewtoolkit.utils.log
 import org.ossreviewtoolkit.utils.logOnce
+import org.ossreviewtoolkit.utils.searchUpwardsForFile
 
-// See https://docs.microsoft.com/en-us/nuget/api/overview.
-private const val SERVICE_INDEX_URL = "https://api.nuget.org/v3/index.json"
-
+private const val DEFAULT_SERVICE_INDEX_URL = "https://api.nuget.org/v3/index.json"
 private const val REGISTRATIONS_BASE_URL_TYPE = "RegistrationsBaseUrl/3.6.0"
 
-class NuGetSupport {
+class NuGetSupport(vararg serviceIndexUrls: String = arrayOf(DEFAULT_SERVICE_INDEX_URL)) {
     companion object {
         val JSON_MAPPER = JsonMapper().registerKotlinModule()
 
         val XML_MAPPER = XmlMapper(XmlFactory().apply {
             // Work-around for https://github.com/FasterXML/jackson-module-kotlin/issues/138.
             xmlTextElementName = "value"
         }).registerKotlinModule()
+
+        fun create(definitionFile: File): NuGetSupport {
+            val configXmlReader = NuGetConfigFileReader()
+            val configFile = definitionFile.parentFile.searchUpwardsForFile("nuget.config", true)
+            val serviceIndexUrls = configFile?.let { configXmlReader.getRegistrationsBaseUrls(it).toTypedArray() }
+
+            return serviceIndexUrls?.let { NuGetSupport(*it) } ?: NuGetSupport()
```

Comment:
If you follow my proposal from above, the upper line can become

            val serviceIndexUrls = configFile?.let { configXmlReader.getRegistrationsBaseUrls(it) }

            return NuGetSupport(serviceIndexUrls.orEmpty())